remotes/fixup_test.go

```diff
@@ -0,0 +1,199 @@
+package remotes
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/remotes"
+	"github.com/deislabs/cnab-go/bundle"
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/manifest/manifestlist"
+	"github.com/docker/distribution/reference"
+	"github.com/opencontainers/go-digest"
+	ocischema "github.com/opencontainers/image-spec/specs-go"
+	ocischemav1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"gotest.tools/assert"
+)
+
+func TestFixupPlatformShortPaths(t *testing.T) {
+	// those cases should not need to fetch any data
+	cases := []struct {
+		name      string
+		platform  string
+		mediaType string
+	}{
+		{
+			name:      "no-filter",
+			mediaType: ocischemav1.MediaTypeImageIndex,
+		},
+		{
+			name:      "oci-image",
+			platform:  "linux/amd64",
+			mediaType: ocischemav1.MediaTypeImageManifest,
+		},
+		{
+			name:      "docker-image",
+			platform:  "linux/amd64",
+			mediaType: images.MediaTypeDockerSchema2Manifest,
+		},
+		{
+			name:      "docker-image-schema1",
+			platform:  "linux/amd64",
+			mediaType: images.MediaTypeDockerSchema1Manifest,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			assert.NilError(t, fixupPlatform(context.Background(), fixupConfig{platform: c.platform}, nil, &imageFixupInfo{
+				resolvedDescriptor: ocischemav1.Descriptor{
+					MediaType: c.mediaType,
+				},
+			}, nil))
```

Comment:
If I read it correctly, these tests only check calling fixupPlatform does not return an error right?

Reply:
It does a little bit more: it checks that sourceFetcher is not used for analyzing the data